.github/workflows/main.yml

```diff
@@ -41,6 +41,13 @@ jobs:
       # only informational so swallow error codes
       - run: npm outdated || exit 0
 
+      # Print .NET info for installed SDK versions
+      - run: dotnet --info
+
+      # Force .NET 6.0.xxx for Blazor build (agent now has a newer version which breaks 'dotnet format', see https://github.com/dotnet/format/issues/1800)
+      - name: Apply global.json for nimble-blazor
+        run: cp packages/nimble-blazor/global.ci.json packages/nimble-blazor/global.json
```

Comment:
This all doesn't feel quite right... You can't have reproduceable dependencies for local projects? Your global installed packages break everything?

Can we setup the lock files to avoid these problems?
https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#locking-dependencies

Looks like lock files could even let us do some caching in the future (though we don't have too many deps):
https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#locking-dependencies

Reply:
@rajsite @atmgrifter00 I don't think PackageReferences directly affect the libraries that `dotnet format` uses/loads. If you look at the docs for [global.json](https://learn.microsoft.com/en-us/dotnet/core/tools/global-json) it says that that's what determines the .NET SDK version used for .NET CLI commands (which is automatically 'latest' unless you override it with a global.json).

Also see [this comment](https://github.com/dotnet/sdk/issues/32598#issuecomment-1554743399) on one of the linked issues:

> dotnet-format should use the same version of the compiler and analyzers that would be used if you just build the project with dotnet build. However, this is currently "impossible" because the SDK/compiler used by the build does not include all of the dependencies that dotnet-format needs to run (e.g. the workspaces layer), and we don't have a distribution strategy in place to provide the matching versions of the required dependencies on demand for every possible combination of SDK a project might be using.
> 
> I'm not even sure how we would provide the matching dependencies if we locked dotnet-format to using the compiler coming from the same SDK that dotnet-format came with. Some of the dependencies are omitted from distribution by long-standing design.

Reply:
So, I think this change with global.json is what we'd need to work around the problem currently, if we want Blazor linting back on immediately.

If we do nothing, I think a future .NET SDK update will have a fix, but I'm not sure how long it would be before that shows up on our GitHub Actions agent. So we could also periodically try to just re-enable linting and see if it fixes itself eventually (whenever that new SDK version comes out).

Reply:
Then do you know what's needed to get the local gobal.json working? If I understand correctly right now it will use whatever is the latest version of dotnet to build?

> It looks like locally I'm using .NET 7.0.202 (and don't have a full install of .NET 6.x, only the SDK), so that global.json doesn't work for me as-is

I don't want to end up in a situation where the formatting / linting behavior is different locally than on CI. It's always frustrating when they give different results...

i.e. why would we only use globals.json on CI, seems like it should be used always?

Reply:
That's basically what I have in "Potential Alternatives" in the PR description, but it's probably missing a few details.
- devs need a .NET 7 SDK locally (i.e. we update contributing docs to say that)
- we update `main.yml` to request .NET 7 specifically (we're already getting the .NET 7 SDK on the agent, but we could specify a more specific version of .NET 7 if we needed to. currently we ask for the .NET 6 SDK)

At that point both local runs and CI runs of `dotnet format` would run against the .NET 7 SDK, and we could use global.json for both local+CI (as both would use .NET 7)

Reply:
I guess this is different than node because with dotnet the runtime is coupled with a bunch of specific libraries (?) where in node everything is an independently updateable library. Node has good backwards compatibility and well controlled library versions via the lock file so doesn't matter as much if I use a newer node version locally.

Sounds like the dotnet toolchain is much more finicky?

Reply:
> At that point both local runs and CI runs of `dotnet format` would run against the .NET 7 SDK, and we could use global.json for both local+CI (as both would use .NET 7)

Do we have to upgrade to .NET 7? We can't lock it at 6?

Reply:
I'm not sure what the consequences of us upgrading to .NET 7 are. Doesn't this introduce a dependency constraint on clients (i.e. force them to upgrade to .NET 7)?

Reply:
More concerned that we don't have control over the .NET version being used locally + on CI. But chatted with Malcolm and I think we can make the global.json apply for both cases and just tell folks to install the .NET SDK we support.

Then when we upgrade to 7 is an independent concern

Reply:
- Now using `global.json` locally too (so removed the copy step for it in the build)
- Updated docs to explicitly call out that a .NET SDK install `6.0.202 <= v < 7` is required. Probably most people already had one (maybe in addition to a .NET 7 SDK) but I didn't.
- Updated the CI build's `setup-dotnet` action to refer to `global.json` to get the required .NET version to install (so we're not encoding the required version in multiple places in code)